scripts/main/header.js

```diff
@@ -479,113 +479,117 @@ header.setMode = function (mode) {
 					".header__toolbar--public, .header__toolbar--albums, .header__toolbar--album, .header__toolbar--map, .header__toolbar--config"
 				)
 			);
-			// If map is disabled, we should hide the icon
-			if (lychee.publicMode === true ? lychee.map_display_public : lychee.map_display) {
-				const e = $("#button_map");
+
+			// Editorial notes:
+			//  a) Keep the order of buttons as defined in the HTML
+			//  b) Only manipulate each button at most once
+			// Otherwise, the code will become completely unmaintainable.
+
+			// These conditions are highly fragile.
+			// The code which stars/unstars a photo assumes that the album
+			// of the photo is loaded, because that code internally uses
+			// `album.getPhotoId`.
+			// (Note, this is surely an error of its own.)
+			// If one attempts to star/unstar a photo and the corresponding
+			// album is not loaded, the code throws an exception.
+			// "By accident" it is safe to assume that the corresponding album
+			// is always loaded if we are not in public mode and the album is
+			// uploadable.
+			// Hence, it is (fortunately) safe to show the button in this case.
+			if (album.isUploadable()) {
+				const e = $("#button_star");
+				e.show();
+			} else {
+				const e = $("#button_star");
+				e.hide();
+			}
+
+			if (lychee.enable_button_visibility && album.isUploadable()) {
+				const e = $("#button_visibility");
 				e.show();
 				tabindex.makeFocusable(e);
 			} else {
-				const e = $("#button_map");
+				const e = $("#button_visibility");
 				e.hide();
 				tabindex.makeUnfocusable(e);
 			}
 
-			if (album.isUploadable()) {
-				const e = $("#button_trash, #button_move, #button_visibility, #button_star, #button_rotate_cwise, #button_rotate_ccwise");
+			if (lychee.enable_button_rotate && album.isUploadable()) {
+				const e = $("#button_rotate_cwise, #button_rotate_ccwise");
 				e.show();
 				tabindex.makeFocusable(e);
 			} else {
-				const e = $("#button_trash, #button_move, #button_visibility, #button_star, #button_rotate_cwise, #button_rotate_ccwise");
+				const e = $("#button_rotate_cwise, #button_rotate_ccwise");
 				e.hide();
 				tabindex.makeUnfocusable(e);
 			}
 
-			if (photo.json && photo.json.hasOwnProperty("is_share_button_visible") && !photo.json.is_share_button_visible) {
+			if (lychee.enable_button_share && photo.json && photo.json.is_share_button_visible) {
 				const e = $("#button_share");
-				e.hide();
+				e.show();
```

Comment:
As you inverted the condition here (which I'm in favor of, for the sake of consistency) you need to switch the order of `makeUnfocusable`/`makeFocusable` as well...

Reply:
Fixed by https://github.com/LycheeOrg/Lychee-front/pull/323/commits/8deda77a8701b6253177823587dcf4f5cbd9eb1c.